internal/config/config_file.go

```diff
@@ -51,30 +52,84 @@ func ConfigDir() string {
 func autoMigrateConfigDir(newPath string) {
 	path, err := os.UserHomeDir()
 	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
-		migrateConfigDir(oldPath, newPath)
+		migrateDir(oldPath, newPath)
 		return
 	}
 
 	path, err = homedir.Dir()
 	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
-		migrateConfigDir(oldPath, newPath)
+		migrateDir(oldPath, newPath)
 	}
 }
 
-func dirExists(path string) bool {
-	f, err := os.Stat(path)
-	return err == nil && f.IsDir()
+func StateDir() string {
+	if path := os.Getenv(XDG_STATE_HOME); path != "" {
+		path = filepath.Join(path, "gh")
+		if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
+			_ = os.MkdirAll(path, 0755)
```

Comment:
This MkdirAll doesn't look like it's necessary here

Reply:
Originally I didn't have it in there, but in the scenario where `XDG_STATE_HOME` is specified and there is no previous `state.yml` file then `XDG_STATE_HOME/gh` does not get created and the code that writes the state file does not create the directory so writing the state file will fail.